packages/postgrest-swr/src/subscribe/use-subscription-query.ts

```diff
@@ -0,0 +1,66 @@
+import { useSWRConfig } from "swr";
+import { useEffect, useId } from "react";
+import {
+  GenericTable,
+  PostgrestSWRMutatorOpts,
+  useCacheScanner,
+  update,
+  insert,
+  remove,
+} from "../lib";
+import { SupabaseClient } from "@supabase/supabase-js";
+import { DEFAULT_SCHEMA_NAME } from "@supabase-cache-helpers/postgrest-shared";
+import { Response, PostgresChangeFilter } from "./types";
+
+function useSubscriptionQuery<T extends GenericTable, Q extends string = "*">(
+  client: SupabaseClient | null,
+  channel: string,
+  filter: PostgresChangeFilter,
+  query: Q,
+  primaryKeys: (keyof T["Row"])[],
+  opts?: Omit<PostgrestSWRMutatorOpts<T>, "schema">
+) {
+  const { mutate } = useSWRConfig();
+  const scan = useCacheScanner(filter.table, opts);
+
+  useEffect(() => {
+    if (!client) return;
+
+    const subscription = client
+      .channel(channel)
+      .on(
+        "postgres_changes",
+        { ...filter, schema: filter.schema ?? DEFAULT_SCHEMA_NAME },
+        async (payload: Response<T>) => {
+          const qb = client.from(filter.table).select(query);
+          for (const pk of primaryKeys) {
+            qb.eq(pk.toString(), (payload.record ?? payload.old_record)[pk]);
```

Comment:
I'm not super familiar with Supabase-JS v2 yet, but I think based on the docs we need to subscribe to the primary keys using a slightly less intuitive mechanism:

```
const channel = supabase
  .channel('db-changes')
  .on(
    'postgres_changes',
    { event: 'UPDATE', schema: 'public', table: 'messages', filter: 'id=eq.1' },
    (payload) => console.log(payload)
  )
  .subscribe()
  ```